plugins/cpu/fu-cpu-device.c

```diff
@@ -95,12 +108,34 @@ fu_cpu_device_convert_vendor (const gchar *vendor)
 	return vendor;
 }
 
+static gboolean
+fu_cpu_is_updatable (GError** error)
+{
+	if (g_access ("/lib/firmware/intel-ucode", R_OK|W_OK) < 0)
+		return FALSE;
+
+	/* This path may not be present on older kernels and wont be supported */
+	if (g_access ("/sys/devices/system/cpu/microcode/reload", R_OK|W_OK) < 0)
+		return FALSE;
+
+	/* check if /boot is writable for initramfs update */
+	if (g_access ("/boot", R_OK|W_OK) < 0)
+		return FALSE;
+
+	return TRUE;
+}
+
 static void
 fu_cpu_device_init (FuCpuDevice *self)
 {
 	fu_device_add_guid_full (FU_DEVICE (self), "cpu",
 				 FU_DEVICE_INSTANCE_FLAG_NO_QUIRKS);
 	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_INTERNAL);
+	/* only Intel supported as of now */
+	if (fu_common_get_cpu_vendor () == FU_CPU_VENDOR_INTEL) {
```

Comment:
This should be any vendor shouldn't it?  Using `/sys/devices/system/cpu/microcode/reload` is the same for all of them?  Maybe checking for that file instead.

Reply:
I don't think we're talking about rocket science here, how about just look for `/lib/firmware/$vendor-ucode` to scale?